src/rp2_common/pico_bootrom/include/pico/bootrom.h

```diff
@@ -60,7 +61,72 @@ void *rom_data_lookup(uint32_t code);
+// The following is a list of bootrom functions available via rom_func_lookup and their signature
+
+typedef uint32_t (*rom_popcount32_fn)(uint32_t);
+#define ROM_FUNC_POPCOUNT32             rom_table_code('P','3')
+
+typedef uint32_t (*rom_reverse32_fn)(uint32_t);
+#define ROM_FUNC_REVERSE32              rom_table_code('R','3')
+
+typedef uint32_t (*rom_clz32_fn)(uint32_t);
+#define ROM_FUNC_CLZ32                  rom_table_code('L','3')
+
+typedef uint32_t (*rom_ctz32_fn)(uint32_t);
+#define ROM_FUNC_CTZ32                  rom_table_code('T','3')
+
+typedef uint8_t *(*rom_memset_fn)(uint8_t *, uint8_t, uint32_t);
+#define ROM_FUNC_MEMSET                 rom_table_code('M','S')
+
+typedef uint32_t *(*rom_memset4_fn)(uint32_t *, uint8_t, uint32_t);
+#define ROM_FUNC_MEMSET4                rom_table_code('S','4')
+
+typedef uint32_t *(*rom_memcpy_fn)(uint8_t *, const uint8_t *, uint32_t);
+#define ROM_FUNC_MEMCPY                 rom_table_code('M','C')
+
+typedef uint32_t *(*rom_memcpy44_fn)(uint32_t *, const uint32_t *, uint32_t);
+#define ROM_FUNC_MEMCPY44               rom_table_code('C','4')
+
+typedef void __attribute__((noreturn)) (*rom_reset_usb_boot_fn)(uint32_t, uint32_t);
+typedef rom_reset_usb_boot_fn reset_usb_boot_fn; // kept for backwards compatibility
+#define ROM_FUNC_RESET_USB_BOOT         rom_table_code('U','B')
+
+typedef void (*rom_connect_internal_flash_fn)(void);
+#define ROM_FUNC_CONNECT_INTERNAL_FLASH rom_table_code('I','F')
+
+typedef void (*rom_flash_exit_xip_fn)(void);
+#define ROM_FUNC_FLASH_EXIT_XIP         rom_table_code('E','X')
+
+typedef void (*rom_flash_range_erase_fn)(uint32_t, size_t, uint32_t, uint8_t);
+#define ROM_FUNC_FLASH_RANGE_ERASE      rom_table_code('R','E')
+
+typedef void (*rom_flash_range_program_fn)(uint32_t, const uint8_t*, size_t);
+#define ROM_FUNC_FLASH_RANGE_PROGRAM    rom_table_code('R','P')
+
+typedef void (*rom_flash_flush_cache_fn)(void);
+#define ROM_FUNC_FLASH_FLUSH_CACHE      rom_table_code('F','C')
+
+typedef void (*rom_flash_enter_cmd_xip_fn)(void);
+#define ROM_FUNC_FLASH_ENTER_CMD_XIP    rom_table_code('C','X')
+
+// Bootrom function: rom_table_lookup
+// Returns the 32 bit pointer into the ROM if found or NULL otherwise.
+typedef void *(*rom_table_lookup_fn)(uint16_t *table, uint32_t code);
+
+// Convert a 16 bit pointer stored at the given rom address into a 32 bit pointer
+#define rom_hword_as_ptr(rom_address) (void *)(uintptr_t)(*(uint16_t *)rom_address)
+
+/*!
+ * \brief Lookup a bootrom function by code. This method is forceably inlined into the caller for FLASH/RAM sensitive code usage
+ * \ingroup pico_bootrom
+ * \param code the code
+ * \return a pointer to the function, or NULL if the code does not match any bootrom function
+ */
+static __force_inline void *rom_func_lookup_inline(uint32_t code) {
+    rom_table_lookup_fn rom_table_lookup = (rom_table_lookup_fn) rom_hword_as_ptr(0x18);
+    uint16_t *func_table = (uint16_t *) rom_hword_as_ptr(0x14);
```

Comment:
Where does the magic `0x14` come from? Is it worth making it a `#define` ? :shrug: 
(Doesn't matter if not, but hopefully it doesn't hurt to ask)

Reply:
it is in the datasheet - dont care to add a define